### ISSUE TITLE
Marathon provider no longer allows for tasks from different apps within the same backend

### ISSUE THREAD

**Issue by TrevinTeacutter** [containous/traefik#2709]
### Do you want to request a *feature* or report a *bug*?

*bug*

### What did you do?

We have the need for new apps to be included within the same backend so that they are added
to the wrr pool of an existing app in order for zero downtime deployments. This was originally
working with Traefik 1.4.0-rc1 and continued to work until 1.4.2. We didn't try to upgrade
until recently and we are pretty sure this template change caused the issue:

https://github.com/containous/traefik/commit/02035d4942c9da9db393739879e66c28d872ce44#diff-080894c4c7f765921235ddb2ceb82140

### What did you expect to see?

We expect to see multiple tasks belonging to different marathon apps within the same backend.

### What did you see instead?

An error due to a backend already being 'configured' within the template

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.4.1 (This is the last version that worked for this)
1.4.2 (First version that failed for the use-case)
1.4.6
1.5.0.rc5

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
################################################################
# Global configuration
################################################################

InsecureSkipVerify = true
defaultEntryPoints = ["http","https"]
email = "test@traefik.io"
storage = "traefik/acme/account"
entryPoint = "https"

[accessLog]

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      CertFile = "/etc/ssl/traefik.crt"
      KeyFile = "/etc/ssl/traefik.key"

[retry]
attempts = -1

[forwardingTimeouts]

# dialTimeout is the amount of time to wait until a connection to a backend server can be established.
#
# Optional
# Default: "30s"
#
dialTimeout = "3s"

################################################################
# Web configuration backend
################################################################

[web]
address = ":9090"

################################################################
# File configuration backend
################################################################
trace = true

################################################################
# Mesos/Marathon configuration backend
################################################################

[marathon]
endpoint = "https://leader.mesos/marathon"
watch = true
domain = "marathon.localhost"
exposedByDefault = true
marathonLBCompatibility = true

[marathon.DCOSCredentials]
UID = "svc-traefik"
PrivateKey = "/etc/ssl/service-account.key"
LoginEndpoint = "https://leader.mesos/acs/api/v1/auth/login"

[marathon.TLS]
InsecureSkipVerify = true
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2018-01-15T18:51:11Z" level=error msg="Failed to render Marathon configuration template: Near line 66 (last key parsed 'backends'): Key 'backends.foo' has already been defined." 

```

**Comment by ldez** [comment]
Related to #2333

**Comment by emilevauge** [comment]
@TrevinTeacutter Could you provide your Marathon application descriptor (json) that produces this error ?

**Comment by TrevinTeacutter** [comment]
```json
{
  "id": "/foo-v000",
  "instances": 2,
  "cpus": 0.1,
  "mem": 32,
  "disk": 0,
  "gpus": 0,
  "constraints": [],
  "fetch": [],
  "storeUrls": [],
  "backoffSeconds": 1,
  "backoffFactor": 1.15,
  "maxLaunchDelaySeconds": 3600,
  "container": {
    "type": "DOCKER",
    "volumes": [],
    "docker": {
      "image": "docker.cernerrepos.net/lwander/spin-kub-demo:yellow",
      "network": "BRIDGE",
      "portMappings": [
        {
          "containerPort": 8080,
          "protocol": "tcp",
          "name": "main",
          "labels": {}
        }
      ],
      "privileged": false,
      "parameters": [],
      "forcePullImage": true
    }
  },
  "healthChecks": [],
  "readinessChecks": [],
  "dependencies": [],
  "upgradeStrategy": {
    "minimumHealthCapacity": 1,
    "maximumOverCapacity": 1
  },
  "secrets": {},
  "unreachableStrategy": {
    "inactiveAfterSeconds": 300,
    "expungeAfterSeconds": 600
  },
  "killSelection": "YOUNGEST_FIRST",
  "requirePorts": false,
  "labels": {
    "traefik.main.backend": "test.foo",
    "traefik.main.protocol": "http",
    "traefik.protocol": "http",
    "traefik.main.portIndex": "0",
    "traefik.enable": "true",
    "traefik.main.frontend.rule": "Host:test.foo.hax"
  },
  "env": {}
}
```
```json
{
  "id": "/foo-v001",
  "instances": 2,
  "cpus": 0.1,
  "mem": 32,
  "disk": 0,
  "gpus": 0,
  "constraints": [],
  "fetch": [],
  "storeUrls": [],
  "backoffSeconds": 1,
  "backoffFactor": 1.15,
  "maxLaunchDelaySeconds": 3600,
  "container": {
    "type": "DOCKER",
    "volumes": [],
    "docker": {
      "image": "docker.cernerrepos.net/lwander/spin-kub-demo:red",
      "network": "BRIDGE",
      "portMappings": [
        {
          "containerPort": 8080,
          "protocol": "tcp",
          "name": "main",
          "labels": {}
        }
      ],
      "privileged": false,
      "parameters": [],
      "forcePullImage": true
    }
  },
  "healthChecks": [],
  "readinessChecks": [],
  "dependencies": [],
  "upgradeStrategy": {
    "minimumHealthCapacity": 1,
    "maximumOverCapacity": 1
  },
  "secrets": {},
  "unreachableStrategy": {
    "inactiveAfterSeconds": 300,
    "expungeAfterSeconds": 600
  },
  "killSelection": "YOUNGEST_FIRST",
  "requirePorts": false,
  "labels": {
    "traefik.main.backend": "test.foo",
    "traefik.main.protocol": "http",
    "traefik.protocol": "http",
    "traefik.main.portIndex": "0",
    "traefik.enable": "true",
    "traefik.main.frontend.rule": "Host:test.foo.hax"
  },
  "env": {}
}
```
We should expect to see this in the Web UI:
<img width="1212" alt="screen shot 2018-01-16 at 9 36 40 am" src="https://user-images.githubusercontent.com/25596866/34997008-e2e76ae6-faa0-11e7-843b-afc0036e324a.png">

**Comment by maximusfloydus** [comment]
Has any done any investigation of this issue?

**Comment by TrevinTeacutter** [comment]
We removed [this line](https://github.com/containous/traefik/commit/02035d4942c9da9db393739879e66c28d872ce44#diff-080894c4c7f765921235ddb2ceb82140R15) in the marathon template in our local fork and this reverted the behavior but to ensure the issue this was added for there would probably need to be smarter logic to group all apps together by backends defined (if any).

**Comment by traefiker** [comment]
Closed by #3109.
